### DISCUSSION TITLE
Possibly different frequency for bootloader and user code?

### DISCUSSION THREAD

**Discussion by TheLastGimbus** [micronucleus/micronucleus#263] [Q&A] [answered]
I wanted to reduce power consumption of my board - one of the options is to reduce the clock speed

Sadly, looks like USB comm (and thus, this bootloader) require 12MHz at least - legit :100:

But I don't plan using usb in my Arduino code tho... would it be *ever* possible to have different clock in bootloader and user code?

**Comment by cpldcpu** [comment, marked as answer]
That is easily possible now. Just change the clockspeed in your user program.

**Reply by TheLastGimbus** [reply]
Looks like i can do this with:
```cpp
#include <avr/power.h>
clock_prescale_set(clock_div_8);
```
But `delay()` is now also 8 times slower :joy: - same as the guy here: https://forum.arduino.cc/t/arduino-1-0-6-attiny-85-delay-much-longer-why/271052

Ps. setting `#define F_CPU 2062500` (16500000/8) at the very top of `main.cpp` file doesn't help at all :disappointed: 

Pps. `millis()` based s

**Reply by cpldcpu** [reply]
seems to be explained here:

https://forum.arduino.cc/t/problem-with-define-f_cpu/455431/3

**Reply by TheLastGimbus** [reply]
Huh... thanks! From all of this, I thought "ok, so maybe I just have to tell platformio to compile *everything* with different frequency :thinking: "

But, if I compile stuff with `board_build.f_cpu` anything other than `16000000` or `8000000`, it panicks and screams (surely `FastLED.h` does)... and micronucleus (without extra crystal) supports only 12.5 and 16.5... so I would have timing *a bit* off, but not as much...

Well, in my case, it may be just not really do-able, but someone else could try this for sure!

**Comment by ArminJo** [comment]
Use e.g. https://github.com/SpenceKonde/ATTinyCore and choose the desired program frequency (F_CPU). The core sets the frequency after the bootloader and before your setup to the right value.

**Reply by TheLastGimbus** [reply]
Hm... as I understand, this is some alternative Arduino core for attiny? Looks interesting :eyes; however, I don't have enought curage to try it now, so can't tell if it woks :/

Could you tell me if this would increase the flash size usage? Or would even decrease it :thinking:  ?
